server/server.go

```diff
@@ -316,10 +330,19 @@ func (s *Server) reload(ctx context.Context, txn storage.Transaction, event stor
 		return
 	}
 
-	if err := s.reloadCompiler(ctx, txn); err != nil {
+	err := s.reloadCompiler(ctx, txn)
+	if err != nil {
+		panic(err)
+	}
+
+	s.watcher, err = s.watcher.Migrate(s.Compiler(), txn)
+	if err != nil {
 		panic(err)
```

Comment:
The compiler reload panics because it should not be possible for compilation to fail here--if policies don't compile, they are not written to storage. Can we make the same assumption about migrate and start?